### ISSUE TITLE
Incompatibility with refactored assets output

### ISSUE THREAD

**Issue by campusboy87** [crstauf/query-monitor-extend#11]
Hey. Thanks for plugin!

Unfortunately, it stopped working. Versions used:
- WordPress 5.0.3
- Any theme
- Query Monitor 3.3.1
- Query Monitor Extend 0.0.5

Screenshot of the main page of **Twenty Seventeen** theme
![screen 20-02-2019 023214](https://user-images.githubusercontent.com/20075333/53055303-3cc9dd80-34b8-11e9-97bd-a2261bd77b41.png)

**Comment by crstauf** [comment]
Thanks for the report, @campusboy87! 

Are there any entries in the server log?

You could also try the `version/1.0` branch. That branch is the most up-to-date, I just haven't merged it into `master` yet.

**Comment by campusboy87** [comment]
Query Monitor Extend 0.0.5 - server log:
```
[20-Feb-2019 00:20:49 UTC] PHP Warning:  Declaration of QMX_Output_Html_Assets::dependency_row(_WP_Dependency $dependency, WP_Dependencies $dependencies, $type) should be compatible with QM_Output_Html_Assets::dependency_row($handle, array $asset, $label) in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\assets.php on line 218
[20-Feb-2019 00:20:49 UTC] PHP Fatal error:  Class QMX_Output_Html_Assets contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (QM_Output_Html_Assets::get_type_labels) in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\assets.php on line 218
```

Removed the old version, installed [version/1.0](https://github.com/crstauf/query-monitor-extend/tree/version/1.0). Error remained :-(. Server log:

```
[20-Feb-2019 00:24:25 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_data() on boolean in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php:62
Stack trace:
#0 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(29): QMX_Admin_Bar->process_and_print_styles()
#1 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(37): QMX_Admin_Bar->__construct()
#2 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(122): QMX_Admin_Bar::get_instance()
#3 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(286): register_qmx_output_html_admin_bar(Array)
#4 F:\server\sites\wp-test.com\wp-includes\plugin.php(203): WP_Hook->apply_filters(Array, Array)
#5 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\classes\QueryMonitorExtend.php(77): apply_filters('qmx/outputter/h...', Array)
#6 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(288): QueryMonitorExtend->filter__qm_outputter_html(Ar in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php on line 62
[20-Feb-2019 00:24:29 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_data() on boolean in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php:62
Stack trace:
#0 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(29): QMX_Admin_Bar->process_and_print_styles()
#1 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(37): QMX_Admin_Bar->__construct()
#2 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(122): QMX_Admin_Bar::get_instance()
#3 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(286): register_qmx_output_html_admin_bar(Array)
#4 F:\server\sites\wp-test.com\wp-includes\plugin.php(203): WP_Hook->apply_filters(Array, Array)
#5 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\classes\QueryMonitorExtend.php(77): apply_filters('qmx/outputter/h...', Array)
#6 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(288): QueryMonitorExtend->filter__qm_outputter_html(Ar in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php on line 62
[20-Feb-2019 00:24:30 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_data() on boolean in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php:62
Stack trace:
#0 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(29): QMX_Admin_Bar->process_and_print_styles()
#1 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(37): QMX_Admin_Bar->__construct()
#2 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(122): QMX_Admin_Bar::get_instance()
#3 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(286): register_qmx_output_html_admin_bar(Array)
#4 F:\server\sites\wp-test.com\wp-includes\plugin.php(203): WP_Hook->apply_filters(Array, Array)
#5 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\classes\QueryMonitorExtend.php(77): apply_filters('qmx/outputter/h...', Array)
#6 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(288): QueryMonitorExtend->filter__qm_outputter_html(Ar in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php on line 62
```

Maybe it's important, I have php 7.2 installed.

**Comment by crstauf** [comment]
Hmmm, thanks for checking so quicklly. I hope to check and fix this tonight.

**Comment by crstauf** [comment]
Mmm, this isn't going to be a quick fix. [QM 3.3 refactored the assets output](https://github.com/johnbillion/query-monitor/commit/d148ad461a0d559c5fe0b3f650204d3a60800a90) to no longer use the `_WP_Dependency` and `_WP_Dependencies` classes. I'll look into this, but I'll likely need to evaluate and refactor as well to support the change in QM.

**Comment by crstauf** [comment]
@campusboy87 In the meantime, please try adding `define( 'QMX_ASSETS_OUTPUT_OVERRIDE', false )` to your `wp-config.php`; that should disable the asset panel override, and (at least) resolve the error you reported.

**Comment by crstauf** [comment]
Considering dropping the assets panel override, as johnbillion/query-monitor#110 has an updated PR, and will hopefully be merged into QM soon.

**Comment by campusboy87** [comment]
> @campusboy87 In the meantime, please try adding `define( 'QMX_ASSETS_OUTPUT_OVERRIDE', false )` to your `wp-config.php`; that should disable the asset panel override, and (at least) resolve the error you reported.

I did it, but it didn't fix the mistake (use [version/1.0](https://github.com/crstauf/query-monitor-extend/tree/version/1.0)). The log like this:

```
[21-Feb-2019 06:10:52 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_data() on boolean in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php:62
Stack trace:
#0 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(29): QMX_Admin_Bar->process_and_print_styles()
#1 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(37): QMX_Admin_Bar->__construct()
#2 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php(122): QMX_Admin_Bar::get_instance()
#3 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(286): register_qmx_output_html_admin_bar(Array)
#4 F:\server\sites\wp-test.com\wp-includes\plugin.php(203): WP_Hook->apply_filters(Array, Array)
#5 F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\classes\QueryMonitorExtend.php(77): apply_filters('qmx/outputter/h...', Array)
#6 F:\server\sites\wp-test.com\wp-includes\class-wp-hook.php(288): QueryMonitorExtend->filter__qm_outputter_html(Ar in F:\server\sites\wp-test.com\wp-content\plugins\query-monitor-extend\output\AdminBar.php on line 62
```

**Comment by crstauf** [comment]
Thanks, @campusboy87; that should be a relatively quick fix. :)

**Comment by crstauf** [comment]
@campusboy87 Fixed in https://github.com/crstauf/query-monitor-extend/commit/30efa8122799771c14f661744aea91cfe82bdac4; will you please test and confirm?

**Comment by campusboy87** [comment]
> @campusboy87 Fixed in [30efa81](https://github.com/crstauf/query-monitor-extend/commit/30efa8122799771c14f661744aea91cfe82bdac4); will you please test and confirm?

I didn't notice any mistakes. Everything works fine. Thank you!
